projects/packages/search/src/dashboard/components/pages/upsell-page.jsx

```diff
@@ -52,15 +52,21 @@ export default function UpsellPage( { isLoading = false } ) {
 		[ fetchSearchPlanInfo ]
 	);
 
-	const { run: sendToCartPaid, hasCheckoutStartedPaid } = useProductCheckoutWorkflow( {
+	const {
+		run: sendToCartPaid,
+		hasCheckoutStarted: hasCheckoutStartedPaid,
+	} = useProductCheckoutWorkflow( {
 		productSlug: 'jetpack_search',
-		redirectUrl: `${ adminUrl }admin.php?page=jetpack-search`,
+		redirectUrl: `${ adminUrl }admin.php?page=jetpack-search&just_upgraded=1`,
 		siteProductAvailabilityHandler: checkSiteHasSearchProduct,
 		from: 'jetpack-search',
 		siteSuffix: domain,
 	} );
 
-	const { run: sendToCartFree, hasCheckoutStartedFree } = useProductCheckoutWorkflow( {
+	const {
+		run: sendToCartFree,
+		hasCheckoutStarted: hasCheckoutStartedFree,
+	} = useProductCheckoutWorkflow( {
 		productSlug: 'jetpack_search_free',
 		redirectUrl: `${ adminUrl }admin.php?page=jetpack-search`,
```

Comment:
Guess we could add `just_upgraded` for free as well?

Reply:
Agreed. I don't think the tooltips support the post-purchase process for free tiers, but it would be nice to have.

Reply:
According to the catch from @a8ck3n, the redirection after the `first` purchased flow (both free and paid) would immediately show the `FirstRunSection` where no tooltips exist. Users must `reload` the dashboard page after indexing without any navigation to see the tooltips, which seems unpractical.

However, we can tackle this enhancement to show the tooltips after indexing is done with extensive work.